python/ray/experimental/client/api.py

```diff
@@ -14,32 +14,92 @@
 
 
 class APIImpl(ABC):
+    """
+    APIImpl is the interface to implement for whichever version of the core
+    Ray API that needs abstracting when run in client mode.
+    """
+
     @abstractmethod
     def get(self, *args, **kwargs):
+        """
+        get is the hook stub passed on to replace `ray.get`
+
+        :param args: opaque arguments
+        :param kwargs: opaque keyword arguments
+        """
         pass
 
     @abstractmethod
     def put(self, *args, **kwargs):
+        """
+        put is the hook stub passed on to replace `ray.put`
+
+        :param args: opaque arguments
+        :param kwargs: opaque keyword arguments
+        """
         pass
 
     @abstractmethod
     def wait(self, *args, **kwargs):
```

Comment:
Nope, on the server side this is Tuple2[List[ObjectRef],...] and on the client side it's as you have it.

Unless you want me to go full `Union[]` of these things, but then they all would have to do that as mypy doesn't handle contravariance/covariance well

Reply:
I think it will make it much easier for other developers to know what type is expected to be returned here if you somehow incorporate (or at the very least comment it)